pkg/synopsysctl/helpers.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright (C) 2019 Synopsys, Inc.
+
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements. See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership. The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied. See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package synopsysctl
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// Flags for using mock mode - doesn't deploy
+var mockFormat string
+var mockKubeFormat = "json"
+
+func addMockKubeFormatFlag(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&mockKubeFormat, "format", "o", mockKubeFormat, "Output format [json|yaml]")
+}
+
+func addMockFlag(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&mockFormat, "mock", mockFormat, "Print the resource spec in the specified format instead of creating it [json|yaml]")
```

Comment:
Why not have mockFormat also default to json?  Could just have a default const that you pass to both Flags() functions that is "json" and value from the option goes into separate variables.

Reply:
@rrati the mock flag is `--mock <string>`. The user must provide a value (json|yaml) for the <string>. We could set the default to json, but when the user sets the flag they are required to put <string> anyways so it will be overwritten. An alternative is to have --mock be a boolean flag and then also have a --format flag that defaults to json.

Reply:
We don't want to change how --mock works, but since --mock requires an arg then no point in do anything related to my comment.